### ISSUE TITLE
WebGLRenderTarget: DepthTexture cannot be changed after initial use

### ISSUE THREAD

**Issue by gkjohnson** [mrdoob/three.js#19447]
I'm using a render target depthTexture to render the main scene, then would like to modify the contents of the color buffer without touching the contents of the depth buffer but it looks like once a render target has been set with a given depthTexture setting it cannot be modified later. Here's a simplified example of what I'm trying to do. In practice I'm using multiple textures and a fullscreen quad to render the second time:

```js
const renderTarget = new WebGLRenderTarget();
const depthTexture = new DepthTexture();

// ...

renderTarget.depthTexture = depthTexture;
renderer.setRenderTarget( renderTarget );
renderer.render( scene, camera );

renderTarget.depthTexture = null;
renderer.setRenderTarget( renderTarget );
renderer.render( secondScene, camera );
```

Looking through the code it looks like these lines are to blame for the depth texture not updating:

https://github.com/mrdoob/three.js/blob/c4d1e885b2bd293f13d6efdfffb0be817845f521/src/renderers/WebGLRenderer.js#L1964-L1968

`textures.setUpRenderTarget` will only get called once per render target, which is [where the depth texture gets initialized and bound to the frame buffer.](https://github.com/mrdoob/three.js/blob/6dbb554d981a0de9043956a6439583b3db293d6a/src/renderers/webgl/WebGLTextures.js#L1135)

**Comment by gkjohnson** [comment]
@Mugen87 @mrdoob are there any existing patterns that can be used to fix this? I was looking into how to address it but there are quite a few corner cases.

Here are the cases:

**depthBuffer = true, depthTexture = null**

- A rendertarget-specific depth buffer should be created.

**depthBuffer = false, depthTexture = null**

- The rendertarget-specific depth buffer should be disposed if it has been created because `depthBuffer` has been set to `false`.

**depthBuffer = true, depthTexture = DepthTexture instance**

- The depthTexture should be bound as the depth buffer and changed if the depthTexture set is different from the last frame.

**depthBuffer = false, depthTexture = DepthTexture instance**

- The depth texture should be unbound but _not_ be disposed.

---

I wonder if something like the [version and needsUpdate](https://github.com/mrdoob/three.js/blob/dev/src/core/BufferAttribute.js#L35-L43) pattern might work for renderTargets, too?

**Comment by gkjohnson** [comment]
FYI for anyone running into this my workaround has been perform a manual copy of the depth textures using the `fragDepth` extension to retain it when running effects on buffers.

**Comment by pailhead** [comment]
I just want to confirm if i understand the example, is what you're trying to achieve possible by turning the depth mask off while rendering to the color attachment?

**Comment by gkjohnson** [comment]
> I just want to confirm if i understand the example, is what you're trying to achieve possible by turning the depth mask off while rendering to the color attachment?

In the above example, yes. But in my work though I was trying to render an effect into another buffer using a fullscreen quad and wanted to move the depthbuffer over so I could draw another mesh without having to do another copy or create a clone of the original depth buffer:

```js
const origTarget = new WebGLRenderTarget( ... );
const effectTarget = new WebGLRenderTarget( ... );
const depthTexture = new DepthTexture();
const effectQuad = new FullScreenQuad( ... );

// ...

origTarget.depthTexture = depthTexture;
renderer.setRenderTarget( origTarget );
renderer.render( scene, camera );

effectQuad.material.uniforms.texture.value = origTarget.texture;
renderer.setRenderTarget( effectTarget );
effectTarget.depthTexture = null;
effectQuad.render( renderer );
effectQuad.depthTexture = depthTexture;

// draw more meshes after the effect has been rendered
```

**Comment by gkjohnson** [comment]
Fixed in #28584
